### ISSUE TITLE
Subawardee_business_type missing in XML? (truncated where present )

### ISSUE THREAD

**Issue by Umer08** [fedspendingtransparency/USASpending-Issue-Tracker#46]
When exporting sub-awards data from the Download Center-->Data Download, Type of Data= Sub-Award, in a CSV export the subawardee_business_type column is populated. Ex: Contracts and Grants~::::~For-Profit Organization~::::~Manufacturer of Goods. 

In the XML this field appears to be missing. Over the last few weeks, I have also noticed that when/where subawardee_business_types is present in the XML, it is truncated (data is missing).

In short, if you can confirm that subawardee_business_type  field is being transmitted in the XML. Second, what is the field name in the XML for this field. Third, please confirm that this field is not incorrectly being truncated. 

Thank you.

**Comment by bsweger** [comment]
@Umer08 Thanks for the report. I did a quick test and saw subawardee_business_type data in the XML download that matched the CSV version (the XML field name is subawardee_business_type).

Can you tell us the exact search you were running when you noticed the missing information (_e.g._, which agency, spending type, time period, and any other options you chose on the _Data Download_ screen). That will help us track down the issue you're seeing.

The quick test that I ran was:
type of data = sub-award
spending type = contracts
agency = Department of Energy
bureau = view all
fiscal year = 2015
recipient state = all
place of performance state = all
date range (obligation date) = 10/1/2014  - (no end date)

**Comment by Umer08** [comment]
@bsweger- Thank you for the response and my apology for the delay on my end! 

So, for some reason, the my XML export does not get structured properly (perhaps this is a reason for why I can't find it). Below are two examples of issues with truncation that a colleague forwarded me (the second half of this ticket). 1. prime award report id: AA6ED218-9508-8A82-F7A0-2B5817F25844 & prime award report id: E181258C-EBDE-8EAE-986C-CA1B9870163. Notice that for both examples, towards the end of the record, the subawardee_business_types gets truncated. 

1) &lt;prime_award_report_id&gt;{AA6ED218-9508-8A82-F7A0-2B5817F25844 &lt;Go&gt;}
&lt;/prime_award_report_id&gt;
&lt;subawardee_business_types&gt;American Indian Owned~::::~Contracts and
Grants~::::~DoT Certified Disadvantaged Busin      ess Enterprise~::::~For-
Profit Organization~::::~Manufacturer of Goods~::::~Minority-Owned
business~::::~Native America      n Owned~::::~Self-Certified Small
Disadvantaged Bu&lt;/subawardee_business_types&gt;

2) &lt;prime_award_report_id&gt;{E181258C-EBDE-8EAE-986C-CA1B9870163F &lt;Go&gt;}
&lt;/prime_award_report_id&gt;
&lt;subawardee_business_types&gt;Asian-Pacific American Owned~::::~Contracts
and Grants~::::~DoT Certified Disadvantage      d Business Enterprise~::::~For-
Profit Organization~::::~Minority-Owned business~::::~SBA Certified 8(a)
Program Partici      pant~::::~Self-Certified Small Disadvantaged Busin&lt;/
subawardee_business_types&gt;


Let me know if you have any questions. I realize this may be a difficult issue to track down (especially because it may involve some ingestion issues on my end). Your cooperation is greatly appreciated!

**Comment by bsweger** [comment]
@Umer08 Thanks for following-up. I can see that the subawardee business type appears truncated in the XML you've provided, but I'm still unable to recreate the circumstances that led to this.

What would help to recreate this issue on our end is some specifics about the search parameters you used to generate the subaward XML download. Could you or the colleague who sent you the XML above tell us the following:

* What agency were you searching?
* Which bureau (if any) were you searching?
* Which fiscal year did you choose?

**Comment by Umer08** [comment]
@bsweger I won't be able to give you the XML till issue #48 is resolved. However, I was able to replicate the issue on the CSV side export. The issue is that the sub_awardee_type field is limited to 255 character length. When the string value goes over, the remaining value is truncated. 

I would share my excel export with you, but I am not sure how do so on Github. Here are the steps.

1) Select type of data= Sub-award
2) Select the spending type= Contracts
3) Filter the Data
       Agency=All
       Bureau= View All
       Fiscal Year= 2015
       Recipient State= All
       PoP State= All
4) Obligation Date Range=01/01/2015
5) Select Type of File= CSV

The exports should be for around 5,119. 

6) Find the subawardee_business_types column.
7) Insert a column to the right or left of it. Title the column "Length". Run the following function in the "length" column. =LEN(CX2) (where Cx2 is the second row of the subawardee_business_types column). 
8) Drag down the values, sort by largest to smallest. You will see that all records that are at 255 are truncated. 

If it is possible to send you the excel, let me know and I can do so. On my end, this is a high priority.

**Comment by bsweger** [comment]
@Umer08 Thanks for detailed steps: I was able to recreate the truncation issue using your directions. The team will investigate, and we'll post updates/fixes here as available.

**Comment by Umer08** [comment]
@bsweger any updates? My technical team is wondering if this can be addressed by tomorrow?

**Comment by bsweger** [comment]
@Umer08 I appreciate your follow-up and apologize that there are no updates to report yet. The team is currently working on some issues related to keyword search; I'll update when there's more to report about the status of the sub-award XML download problems.

**Comment by USAspendingalice** [comment]
@Umer08 The tech team is still looking at this issue to see how to expand the field to include more than the 255 characters.  We appreciate your patience.

**Comment by USAspendingalice** [comment]
This has been resolved.

**Comment by Umer08** [comment]
@USAspendingalice Thank you! Your work is greatly appreciated.
